src/apps/chahub/utils.py

```diff
@@ -0,0 +1,48 @@
+from django.conf import settings
+
+import logging
+import requests
+import json
+
+logger = logging.getLogger(__name__)
+
+
+def send_to_chahub(endpoint, data, update=False):
+    """
+    Does a post request to the specified API endpoint on chahub with the inputted data.
+    :param endpoint: String designating which API endpoint; IE: 'producers/'
+    :param data: Dictionary containing data we are sending away to the endpoint.
+    :return:
+    """
+    assert endpoint, Exception("No ChaHub API endpoint given")
+    assert settings.CHAHUB_API_URL, "CHAHUB_API_URL env var required to send to Chahub "
+
+    url = "{}{}".format(settings.CHAHUB_API_URL, endpoint)
+
+    data = json.dumps(data)
+
+    logger.info("ChaHub :: Sending to ChaHub ({}) the following data: \n{}".format(url, data))
+    try:
+        kwargs = {
+            'url': url,
+            'headers': {
+                'Content-type': 'application/json',
+                'X-CHAHUB-API-KEY': settings.CHAHUB_API_KEY,
+            }
+        }
+        if update:
+            # return requests.patch(data=data, **kwargs)
+            return requests.put(data=data, **kwargs)
+        else:
+            return requests.post(data=data, **kwargs)
+    except requests.ConnectionError:
+        return None
```

Comment:
We should probably return a more accurate error message than simply `None`. And we are trying to catch this error twice (that I have seen so far), so in the second place we won't actually get the error message.
see: src/apps/chahub/tasks.py  Line 22.
Let's add this as a `# TODO:` here. I'll plan on addressing it when I start re-writing tests.

Reply:
Added two do's on both ends for now. We need to come up with a way to standardize that so we all know which errors should bubble out, and which should not.